src/cryptsetup/cryptsetup-keyfile.c

```diff
@@ -1,110 +1,57 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
-#include <unistd.h>
-
 #include "cryptsetup-keyfile.h"
-#include "fd-util.h"
-#include "format-util.h"
-#include "memory-util.h"
+#include "fileio.h"
 #include "path-util.h"
-#include "stat-util.h"
 #include "strv.h"
 
-#define KEY_FILE_SIZE_MAX (16U*1024U*1024U) /* 16 MiB */
-
-int load_key_file(
+int search_key_file(
                 const char *key_file,
                 char **search_path,
-                size_t key_file_size,
-                uint64_t key_file_offset,
+                const char *bindname,
                 void **ret_key,
                 size_t *ret_key_size) {
 
-        _cleanup_(erase_and_freep) char *buffer = NULL;
-        _cleanup_free_ char *discovered_path = NULL;
-        _cleanup_close_ int fd = -1;
-        ssize_t n;
+        char **i;
         int r;
 
         assert(key_file);
         assert(ret_key);
         assert(ret_key_size);
 
         if (strv_isempty(search_path) || path_is_absolute(key_file)) {
```

Comment:
There's just one caller that always specifies a search path. So this code might be simplified...

Reply:
true. I wrote this code also to prepare things for the credentials logic (so that you can supply key data via LoadCredential= too), but didn't include the patch for that yet. In that case we'd not specify a search path. I'd thus like to keep this in for now. will post patch for creds hookup shortly